src/Testcontainers.Db2/Db2Builder.cs

```diff
@@ -0,0 +1,202 @@
+            .WithEnvironment("DB2INST1_PASSWORD", password);
+    }
+
+    /// <summary>
+    /// Sets the Db2 archive logs.
+    /// </summary>
+    /// <param name="archiveLogs">The Db2 archive logs setting.</param>
+    /// <returns>A configured instance of <see cref="Db2Builder" />.</returns>
+    public Db2Builder WithArchiveLogs(bool archiveLogs)
+    {
+        return Merge(DockerResourceConfiguration, new Db2Configuration(archiveLogs: archiveLogs))
+            .WithEnvironment("ARCHIVE_LOGS", archiveLogs.ToString());
+    }
+
+    /// <summary>
+    /// Sets the Db2 autoconfig setting.
+    /// </summary>
+    /// <param name="autoConfig">The Db2 autoconfig setting.</param>
+    /// <returns>A configured instance of <see cref="Db2Builder" />.</returns>
+    public Db2Builder WithAutoconfig(bool autoConfig)
+    {
+        return Merge(DockerResourceConfiguration, new Db2Configuration(autoConfig: autoConfig))
+            .WithEnvironment("AUTOCONFIG", autoConfig.ToString());
+    }
+
+    /// <summary>
+    /// Accepts the Db2 license agreement.
+    /// </summary>
+    /// <returns>A configured instance of <see cref="Db2Builder" />.</returns>
+    public Db2Builder WithLicenseAgreement()
+    {
+        return Merge(DockerResourceConfiguration, new Db2Configuration(licenseAgreement: DefaultLicenseAgreement))
+            .WithEnvironment("LICENSE", DefaultLicenseAgreement);
+    }
+
+    /// <summary>
+    /// Maps the database to memory.
+    /// </summary>
+    /// <returns>A configured instance of <see cref="Db2Builder" />.</returns>
+    public Db2Builder WithInMemoryDatabase()
+    {
+        return Merge(DockerResourceConfiguration, new Db2Configuration(licenseAgreement: DefaultInMemoryDatabasePath))
+            .WithTmpfsMount(DefaultInMemoryDatabasePath);
+    }
+
+    /// <inheritdoc />
+    public override Db2Container Build()
+    {
+        Validate();
+
+        // By default, the base builder waits until the container is running. However, for Db2, a more advanced waiting strategy is necessary
+        // If the user does not provide a custom waiting strategy, append the default Db2 waiting strategy.
+        var db2Builder = DockerResourceConfiguration.WaitStrategies.Count() > 1
+            ? this
+            : WithWaitStrategy(Wait.ForUnixContainer()
+                .UntilMessageIsLogged("All databases are now active")
+                .UntilMessageIsLogged("Setup has completed.")
+                .AddCustomWaitStrategy(new WaitUntil(DockerResourceConfiguration))
+            );
+
+        return new Db2Container(db2Builder.DockerResourceConfiguration);
+    }
+
+    /// <inheritdoc />
+    protected override Db2Builder Init() => base.Init()
+        .WithImage(Db2Image)
+        .WithPortBinding(Db2Port, true)
+        .WithDatabase(DefaultDatabase)
+        .WithUsername(DefaultUsername)
+        .WithPassword(DefaultPassword)
+        .WithLicenseAgreement()
```

Comment:
Can we please use the same pattern as we do in the ServiceBus module: [1)](https://github.com/testcontainers/testcontainers-dotnet/blob/888da44936457c0d3dea0d149e4fb61f23544b33/src/Testcontainers.ServiceBus/ServiceBusBuilder.cs#L43-L55), [2)](https://github.com/testcontainers/testcontainers-dotnet/blob/888da44936457c0d3dea0d149e4fb61f23544b33/src/Testcontainers.ServiceBus/ServiceBusBuilder.cs#L64-L76)?

Reply:
Will adjust

Reply:
I have just seen that I also have to adjust this part. Will fix it in the next few days.